soc/silabs/silabs_s2/soc.c

```diff
@@ -39,3 +52,53 @@ void soc_early_init_hook(void)
 	sl_hfxo_manager_init();
 #endif
 }
+
+#if defined(CONFIG_ARM_SECURE_FIRMWARE) && !defined(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
+static void smu_fault(void)
+{
+	PR_FAULT_INFO("***** SMU FAULT *****");
+
+	if (SMU->IF & SMU_IF_BMPUSEC) {
+		PR_FAULT_INFO("Bus Manager Fault");
+		PR_EXC("SMU.BMPUFS=%d", SMU->BMPUFS);
+	}
+	if (SMU->IF & SMU_IF_PPUSEC) {
+		PR_FAULT_INFO("Peripheral Access Fault");
+		PR_EXC("SMU.PPUFS=%d", SMU->PPUFS);
+	}
+
+	z_fatal_error(K_ERR_CPU_EXCEPTION, NULL);
+}
+#endif
+
+void soc_prep_hook(void)
+{
+	/* Initialize TrustZone state of the device.
+	 * If this is a secure app with no non-secure callable functions, it is a secure-only app.
+	 * Configure all peripherals except the SMU and SEMAILBOX to non-secure aliases, and make
+	 * all bus transactions from the CPU have non-secure attribution.
+	 * This makes the secure-only app behave more like a non-secure app, allowing the use of
+	 * libraries that only expect to use non-secure peripherals, such as the radio subsystem.
+	 */
+#if defined(CONFIG_ARM_SECURE_FIRMWARE) && !defined(CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS)
+	CMU_S->CLKEN1_SET = CMU_CLKEN1_SMU;
+
+	SMU->PPUSATD0_CLR = _SMU_PPUSATD0_MASK;
+#if defined(SEMAILBOX_PRESENT)
```

Comment:
`if (IS_ENABLED(SEMAILBOX_PRESENT))` ...

Reply:
Won't work, `SMU_PPUSATD1_SEMAILBOX` is only available on devices with hardware SE, so it won't compile on BG22 or BG27.